### ISSUE TITLE
WcsNDMap.make_cutout width option

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1608]
I noticed this:
```python
>>> import astropy.units as u
>>> from gammapy.maps import Map
>>> m = Map.create()
>>> m.make_cutout(position=(0, 0), width=(20, 10) * u.deg, mode='trim')[0].geom
WcsGeom

	npix      : 50 x 100 pix
	coordsys  : CEL
	projection: CAR
	center    : 177.5 deg, -85.0 deg
	width     : 5.0 x 10.0 deg
	ndim      : 2
	axes      : 

```

The docstring of make_cutout says that the width is (width_lon, width_lat):
> Angular sizes of the region in (lon, lat). If only one value is passed, a square region is extracted. For more options see also Cutout2D.

http://docs.gammapy.org/dev/api/gammapy.maps.WcsNDMap.html#gammapy.maps.WcsNDMap.make_cutout

But really the order is (width_lat, width_lon), becuase that's how
http://docs.astropy.org/en/latest/api/astropy.nddata.utils.Cutout2D.html
treats the `size` option.

@adonath @registerrier @AtreyeeS - I think we should change the implementation to use `width_lon, width_lat` here, because that's then more consistent with the rest of the gammapy.maps API?

**Comment by cdeil** [comment]
There's two other changes I think we should consider:

1. Change the default mode to "trim" or "partial", I think most users prefer to get something and "strict" is not the expected default behaviour for them.

2. Not return the slices here. Users don't understand what those are, why they get a tuple instead of just a map. Roberta just asked me that after using it for the first time.

Experts still need the slices of course, but those can be behind a second expert method, on the map or  geom object, like e.g. `map.geom.cutout_slices`

@adonath @registerrier - Thoughts?

**Comment by AtreyeeS** [comment]
Hello,

Firstly, I am sorry about the mistype in the doc string!

Regarding the two other points, I agree we can change the default cutout mode. Maybe we keep it `trim`, which is the default astropy value?

My original PR (#1446) for making the cutout did not return the cutout slices precisely because I thought it might be confusing. However, while changing `MapMaker` during the coding sprint to use this `make_cutout` instead of the old one (which was only for 3D maps), I felt it was quicker to stick to returning the cutout slices

`exclusion_mask_cutout, cutout_slices = self.exclusion_map.make_cutout(
                 obs.pointing_radec, 2 * self.offset_max, mode=self.cutout_mode`

Since we wanted a quick working 3D example on the penultimate day, I did not want to spend time thinking of a better implementation. But since `MapMaker` is also going through changes now, I think it is better to clean it up now.

**Comment by cdeil** [comment]
I would suggest to add something like cutout_slices to geom and then use that from the MapMaker.
And then the map.make_cutout would call self.geom.cutout_slices and apply them, but not return them. And yet, +1 to default "trim" for map.make_cutout.

**Comment by registerrier** [comment]
Shall I do it? Or do you do it yourself @cdeil ? There are several tests that use `make_cutout()[0]` which will need to be updated.

**Comment by cdeil** [comment]
I'll do this later today.

**Comment by cdeil** [comment]
I wanted to do this yesterday evening, but noticed that trying to handle `width` properly is difficult.

The first issue is that for `width` can be passed to `Map.create`, e.g. `width=(10, 5)` to make a map that's approximately 10 deg in lon and 5 deg in lat. That parameter is missing from the docstring [here](http://docs.gammapy.org/dev/api/gammapy.maps.Map.html#gammapy.maps.Map.create), but it's there and it has order (lon, lat).

So if we want consistency and also order (lon, lat) for cutout, and keep using Cutout2D from Astropy, we have to switch to (lat, lon) before passing it there. Note how complicated the width handling is there:
https://github.com/astropy/astropy/blob/f84a7a10527a341dd003a30baa8f07f9affa658f/astropy/nddata/utils.py#L663-L664

We have to think what we want to accept and make it work for all those cases (established via tests), both for Map.create and make_cutout and then they have to behave in the same way. E.g.
`width=(10, 5)`, `width=(Angle('10 deg'), Angle('5 deg'))`, `width=Angle([10, 5], 'deg')` are things we might want to support, and also maybe `width=10`, width=Angle('10 deg')`. So input validation / handling of width into a uniform format probably needs a utility function, and then use in multiple places.

Finally I noticed that we have a crop:
https://github.com/gammapy/gammapy/blob/e7b90c1a3b85721168ef7e4fc328a872fd5feec4/gammapy/maps/wcsnd.py#L421-L424

It has no docstring and no good test.

We should probably not keep both crop and cutout, or if we do, have a reasoning and docstring / test for both.

@adonath - given you wrote crop / cutout for SkyImage and this is very complex, I'm hoping you will take this on as main remaining task for this week?

**Comment by cdeil** [comment]
Done by @adonath in #1655 . Closing.

**Comment by cdeil** [comment]
I'm re-opening this issue. See #1673 by @registerrier - apparently order is still lat / lon.

It's agreed order for width in Gammapy should be consistently lon / lat, no?

@adonath - Thoughts? Can you make it better?

**Comment by adonath** [comment]
The issue was not yet fixed in #1655. I think the solution is to split out the width handling to a little helper function, that is called from both `Map.cutout()` as well as `Map.create()`. This way we ensure consistency and minimal code duplication.

**Comment by cdeil** [comment]
Done in #1674 .
